internal/bminventory/inventory.go

```diff
@@ -491,10 +491,22 @@ func (b *bareMetalInventory) RegisterClusterInternal(
 	url := installer.V2GetClusterURL{ClusterID: id}
 
 	log := logutil.FromContext(ctx, b.log).WithField(ctxparams.ClusterId, id)
-	log.Infof("Register cluster: %s with id %s and params %+v", swag.StringValue(params.NewClusterParams.Name), id, params.NewClusterParams)
+
+	// Masked the Pull Secret to prevent its value from being displayed in the logs.
+	clusterParamsNoPullSecret := *params.NewClusterParams
+	if clusterParamsNoPullSecret.PullSecret != nil {
+		*clusterParamsNoPullSecret.PullSecret = "***"
+	}
+
+	jsonNewClusterParams, err := json.Marshal(clusterParamsNoPullSecret)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I think it is better to move this code to external method. This function is already long enough

Reply:
Due to the error that requires handling, I can't execute it within the log statement, I must execute it beforehand to handle the error, So, I'm not sure if it's worth moving to an external function - what do you think?